### PR TITLE
Perplexity Metric Card

### PR DESCRIPTION
**Pull request by emibaylor**
Add Perplexity metric card

Note that it is currently still missing the citation, but I plan to add it later today.

### CODE REVIEW

**Comment by sashavor** [comment]
Otherwise, looks good! Good job, @emibaylor !

File: metrics/perplexity/README.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+## Intended Uses
+Any language generation task.
+
+## How to Use
+
+```python
+from datasets import load_metric
+exact_match_metric = load_metric("perplexity")
+results = exact_match_metric.compute(predictions=predictions, references=references)
```
Comment (lhoestq) [inline metrics/perplexity/README.md:14]: This doesn't look perplexity :P

**Review by lhoestq — COMMENTED** [review]
Cool thanks !

> I'm wondering if we should add that perplexity can be used for analyzing datasets as well

Yes I think it can totally be explained here :)

File: metrics/perplexity/README.md (hunk 1 above)
Comment (emibaylor) [reply metrics/perplexity/README.md:14]: Whoops!! You're right, sorry about that! Will fix right now

File: metrics/perplexity/README.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+1. to evaluate how well the model has learned the distribution of the text it was trained on
+    - In this case, the model input should be the trained model to be evaluated, and the input texts should be the text that the model was trained on.
+2. to evaluate how well a selection of text matches the distribution of text that the input model was trained on
+    - In this case, the model input should be a trained model, and the input texts should be the text to be evaluated.
+
+## Intended Uses
+Any language generation task.
+
+## How to Use
+
```
Comment (lhoestq) [inline metrics/perplexity/README.md:14]: As in the other metric cards, you can add this here:
````suggestion

The metric takes a list of text as input, as well as the name of the model used to compute the metric:

```python
from datasets import load_metric
perplexity = load_metric("perplexity")
results = perplexity.compute(input_texts=input_texts, model_id='gpt2')
```

````

File: metrics/perplexity/README.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    @article{jelinek1977perplexity,
+    title={Perplexity—a measure of the difficulty of speech recognition tasks},
+    author={Jelinek, Fred and Mercer, Robert L and Bahl, Lalit R and Baker, James K},
+    journal={The Journal of the Acoustical Society of America},
+    volume={62},
+    number={S1},
+    pages={S63--S63},
+    year={1977},
+    publisher={Acoustical Society of America}
+    }
```
Comment (lhoestq) [inline metrics/perplexity/README.md:74]: You can use a bibtex block here:

````suggestion
```bibtex
@article{jelinek1977perplexity,
title={Perplexity—a measure of the difficulty of speech recognition tasks},
author={Jelinek, Fred and Mercer, Robert L and Bahl, Lalit R and Baker, James K},
journal={The Journal of the Acoustical Society of America},
volume={62},
number={S1},
pages={S63--S63},
year={1977},
publisher={Acoustical Society of America}
}
```
````

**Review by lhoestq — APPROVED** [review]
Cool thank you ! I've added some minor suggestions, feel free to merge if they sounds good to you

**Pull request merged**